photon-client/src/main/scala/com/linkedin/photon/ml/util/Utils.scala

```diff
@@ -294,7 +292,7 @@ protected[ml] object Utils {
   @throws(classOf[IllegalArgumentException])
   def getKeyFromMapOrElse[T](map: Map[String, Any], key: String, elseBranch: Either[String, T]): T = {
     map.get(key) match {
-      case Some(x: T) => x
+      case Some(x: Any) => x.asInstanceOf[T] // to avoid type erasure warning
```

Comment:
On the other hand, the overhead to using type tags in this case seems prohibitive. Let's keep the warning :-/